src/utils/data-access.ts

```diff
@@ -0,0 +1,98 @@
+import { PromiseResult } from 'aws-sdk/lib/request';
+import { AWSError } from 'aws-sdk/lib/error';
+import DynamoDB, { DocumentClient, QueryOutput } from 'aws-sdk/clients/dynamodb';
+import { Configurations } from './configuration';
+import * as domain from '../domain';
+
+export class DataAccess {
+  private static instance: DataAccess;
+
+  private readonly tableName: string = Configurations.getInstance().dynamoTableName;
+
+  private static docClient: DocumentClient;
+
+  private constructor() {
+    console.log('config', Configurations.getInstance().dynamoTableName);
+    DataAccess.docClient = new DynamoDB.DocumentClient(Configurations.getInstance().dynamoParams);
+  }
+
+  public static getInstance(): DataAccess {
+    if (!DataAccess.instance) {
+      DataAccess.instance = new DataAccess();
+    }
+
+    return DataAccess.instance;
+  }
+
+  public getById(options: {
+    indexName: string;
+    keyCondition: string;
+    expressionAttributeValues: { [key: string]: string };
+  }): Promise<PromiseResult<QueryOutput, AWSError>> {
+    const params = {
+      TableName: this.tableName,
+      IndexName: options.indexName,
+      KeyConditionExpression: options.keyCondition,
+      ExpressionAttributeValues: options.expressionAttributeValues,
+    };
+    console.log('params getbyId', params);
+    return DataAccess.docClient.query(params).promise();
+  }
+
+  public async put<T>(
```

Comment:
Should this be private? Mock it in the unit tests.

Reply:
Done